architecting-cloud-native-dotnet-apps-for-azure/observability-patterns.md

```diff
@@ -3,12 +3,57 @@ title: Observability patterns
+
+## Monitoring
+
+Having centralized logging for searching for issues is helpful in the event that a problem is discovered in the application. In an ideal world, however, end users should not be the ones reporting errors. By taking a proactive approach to failure detection problems can be detected and corrected before the end users are even aware that there has been an issue.
+
+Some centralized logging systems take on an additional role of being able to collect telemetry outside of pure logs. They can collect metrics such as time to run an SQL query, average response time from a web server, even load averages and memory pressure as reported by the operating system. In conjunction with the logs these systems can provide a holistic view of the health of nodes in the system and the application as a whole. 
+
+The metric gathering capabilities of the monitoring tools can also be fed manually from within the application. Business flows which are of particular interest such as new users signing up or orders being placed, may be instrumented such that they increment a counter in the central monitoring system. this unlocks the monitoring tools to not only monitor the health of the application but the health of the business.
+
+Queries can be constructed in the log aggregation tools to look for certain statistics or patterns which can then be displayed, in graphical form, on bespoke dashboards. Frequently teams will invest in large, wall mounted, displays which rotate through the statics about an application. In this way it is very simple to see problems as they are occurring. 
+
+## Alerts
+
+The logical next step is improving reactions to live site incidents is to craft queries against the monitoring tools to look for known failure conditions. For instance, queries could search through the incoming logs for indications of status code 500, which indicates a problem on a web server, being returned. As soon as one of these is detected then an e-mail or an SMS could be sent to the owner of the originating service who can begin to investigate. 
+
+Typically, though, a single 500 error is not sufficient to determine that a problem has occurred. It could simply be that a user mistyped their password or entered some malformed data. The alert queries can be crafted to only fire when a larger than average number of 500 errors are detected. 
+
+One of the most damaging patterns in alerting is to fire too many alerts for humans to investigate. Service owners will rapidly become desensitized to errors which they’re previously investigated and found to be benign. When true errors occur then they will be lost in the noise of hundreds of false positives. This makes it important to ensure that the alerts which do fire are indicative of a real problem.
```

Comment:
which they're previously -> which they've previously

perhaps worth a reference to 'the boy who cried wolf'

Reply:
You mean the non-gender specific child?